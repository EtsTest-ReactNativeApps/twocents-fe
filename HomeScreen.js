
import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2C2540',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: '#EE933F',
    fontSize: 30,
  }
});

export class HomeScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Home!</Text>
      </View>
    );
  }
}