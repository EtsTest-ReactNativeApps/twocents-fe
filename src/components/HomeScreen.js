import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import SearchBar from './SearchBar';

export class HomeScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <SearchBar />
        <Text style={styles.text}>Home!</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2C2540',
    alignItems: 'center',
    paddingTop: 50
  },
  text: {
    color: '#EE933F',
    fontSize: 30,
  }
});