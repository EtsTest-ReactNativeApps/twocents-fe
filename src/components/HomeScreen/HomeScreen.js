import React, { Component } from 'react';
import { Text, View, StyleSheet, ScrollView } from 'react-native';
import { getAllRecommendations } from '../../../utils/APICalls'
import { CategoryScroll } from '../CategoryScroll/CategoryScroll'
import { connect } from 'react-redux';
import { setRecommendations } from '../../actions/index';


export class HomeScreen extends Component {
  constructor(props) {
    super(props)
    this.state = {
      LoggedIn: false,
      categories: []
    }
  }

  static navigationOptions = {
    headerTintColor: '#EE933F'
  }

  componentDidMount() {
    this.fetchRecommendations()
  }

  fetchRecommendations = async () => {
    await getAllRecommendations()
      .then(data => this.props.setRecommendations(data))
      console.log(this.props.allRecommendations[0])
    this.filterCategories()
  }

  filterCategories = () => {
    const categoryList = this.props.allRecommendations.reduce((acc, rec) => {
      rec.categories.forEach(cat => {
        if(!acc[cat]) {
          acc[cat] = []
        }
        acc[cat].push(rec)
      })
      return acc
    }, {})
    this.setState({categories: categoryList})
  }
  
  render() {
    const sideScroll = Object.entries(this.state.categories).map((cat, key) => (
      <CategoryScroll 
        category={cat[0]} 
        recommendations={cat[1]} 
        key={key}
        navigation={this.props.navigation}  
      />
    ))
    return (
      <View style={styles.container}>
        <ScrollView  style={styles.scroll}>
          {sideScroll}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2C2540',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    
  },
  scroll: {
    flex: 1,
    backgroundColor: '#2C2540',
    height: '100%',
    width: '100%'
  },
  text: {
    color: '#EE933F',
    fontSize: 30,
  }
});

const mapStateToProps = state => ({
  allRecommendations: state.allRecommendations
})

const MapDispatchToProps = dispatch => ({
  setRecommendations: (cat) => dispatch(setRecommendations(cat))
})

export default connect(mapStateToProps, MapDispatchToProps)(HomeScreen)