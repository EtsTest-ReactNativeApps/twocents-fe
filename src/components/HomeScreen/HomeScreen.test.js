import React from 'react';
import { shallow } from 'enzyme';
import { HomeScreen, mapStateToProps, mapDispatchToProps } from './HomeScreen';
// import {NavigationTestUtils} from "react-navigation/NavigationTestUtils";

describe('HomeScreen', () => {
  // NavigationTestUtils.resetInternalState();
  let wrapper, instance;
  const mockGeolocator = {
    getCurrentPosition: jest.fn()
  };
  const createTestProps = props => ({
    navigation: {
      navigate: jest.fn()
    },
    allRecommendations: [],
    ...props
  })

  beforeEach(() => {
    global.navigator.geolocation = mockGeolocator;
    props = createTestProps({});
    wrapper = shallow(<HomeScreen {...props} />);
  })

  it('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  xit('should call fetchRecommendations', () => {
    jest.spyOn(instance, "fetchRecommendations");
    expect(instance.fetchRecommendations).toHaveBeenCalledTimes(0)
    instance.componentDidMount()
    expect(instance.fetchRecommendations).toHaveBeenCalledTimes(1)
  })

  describe('mapStateToProps', () => {

    it('allRecommendations should have a state of allRecommendations', () => {
      const mockState = {
        allRecommendations: 'allRecommendations'
      }
      const expected = {
        allRecommendations: 'allRecommendations'
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('userLocation should have a state of userLocation', () => {
      const mockState = {
        userLocation: 'userLocation'
      }
      const expected = {
        userLocation: 'userLocation'
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('loggedIn should have a state of loggedIn', () => {
      const mockState = {
        loggedIn: 'loggedIn'
      }
      const expected = {
        loggedIn: 'loggedIn'
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
  
});