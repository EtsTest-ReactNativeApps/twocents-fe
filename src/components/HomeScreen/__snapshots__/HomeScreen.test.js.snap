// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2C2540",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "#2C2540",
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EE933F",
          "fontSize": 30,
          "marginTop": "70%",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      You have no recommendations saved yet, search for some!
    </Text>
  </ScrollViewMock>
</View>
`;
